hedera-node/src/main/java/com/hedera/services/context/SideEffectsTracker.java

```diff
@@ -354,9 +359,9 @@ public void trackNftOwnerChange(final NftId nftId, final AccountID from, Account
 	 *
 	 * @return the ordered net balance changes
 	 */
-	public TransferList getNetTrackedHbarChanges() {
-		purgeZeroAdjustments(netHbarChanges);
-		return netHbarChanges.build();
+	public CurrencyAdjustments getNetTrackedHbarChanges() {
+		numHbarBalancesChangedSoFar = purgeZeroChanges(changedAccounts, balanceChanges, numHbarBalancesChangedSoFar);
+		return CurrencyAdjustments.fromChanges(balanceChanges, changedAccounts, numHbarBalancesChangedSoFar);
```

Comment:
Instead of requiring `CurrencyAdjustments.fromChanges()` to know how to interpret `numHbarBalancesChangedSoFar`, imo we should do the `Arrays.copyOfRange()` calls here.

(Also note it would be a disaster if the `CurrencyAdjustments` object referenced the **same arrays used by `SideEffectsTracker`**; so it's safest to only pass a copied array here.)

Reply:
Thanks @tinker-michaelj  !! 🙂  Passed the copied arrays into `CurrencyAdjustments`